utoipa-gen/src/schema/component.rs

```diff
@@ -795,13 +795,8 @@ where
                         if component_part.is_any() {
                             tokens.extend(quote! { utoipa::openapi::ObjectBuilder::new() })
                         } else if is_inline {
-                            let assert_component = format_ident!("_Assert{}", name);
                             tokens.extend(quote_spanned! {component_ident.span()=>
-                                {
-                                    struct #assert_component where #component_ident : utoipa::Component;
-                                    
-                                    <#component_ident as utoipa::Component>::component()
-                                }
+                                <#component_ident as utoipa::Component>::component()
```

Comment:
Same here as above, the assert stuct was there to provide better errors for users which is also a pattern demonstrated in syn crate's docs.

Reply:
https://github.com/juhaku/utoipa/pull/215#discussion_r921544433

Reply:
We can remove these assert structs then. Less code anyway better :smile: